api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -493,15 +488,33 @@ private User initializeUserIfNeeded() {
   @Override
   public ResponseEntity<Profile> syncTrainingStatus() {
     User user = userProvider.get();
-    Profile profile = profileService.getProfile(user);
     try {
       userService.syncUserTraining(user);
     } catch(NotFoundException ex) {
       throw ex;
     } catch (ApiException e) {
       throw new ServerErrorException(e);
     }
-    return ResponseEntity.ok(profile);
+    return getProfileResponse(user);
+  }
+
+  @Override
+  public ResponseEntity<Profile> syncEraCommonsStatus() {
+    User user = userProvider.get();
+    NihStatus nihStatus = fireCloudService.getNihStatus();
+    if (nihStatus != null) {
+      if ((nihStatus.getLinkedNihUsername() != null &&
+          nihStatus.getLinkedNihUsername() != user.getEraCommonsLinkedNihUsername()) ||
+          nihStatus.getLinkExpireTime() != user.getEraCommonsLinkExpireTime().getTime()) {
+        user.setEraCommonsCompletionTime(new Timestamp(clock.instant().toEpochMilli()));
+      }
+      user.setEraCommonsLinkedNihUsername(nihStatus.getLinkedNihUsername());
```

Comment:
What should be the completion time if linkedNihUsername is null?

Reply:
If linkedNihUsername is null, completion time should be null. I was acting under the assumption that shibboleth will return an error if there is no username? But don't know if that's correct in truth.

Reply:
Shibboleth doesn't return errors, per se. Remember the flow is:
* AoU -> Shibboleth -> NIH
* User authenticates, or fails to authenticate
  - If the user fails to auth, NIH doesn't redirect back to Shibboleth
* NIH -> Shibboleth -> AoU with a (theoretically) valid token
* AoU -> Post token to FC -> AoU with NIH Status information.